main/sceneManager.cpp

```diff
@@ -1,26 +1,68 @@
 #include "main/sceneManager.h"
 
-using Tri = objects::TriangleObj;
+// TODO: change underscore to the end
 
-void IamSimulator()
+SceneManager::SceneManager()
 {
-    SceneManager manager;
-    CHECK(manager.loadModel("123231")) << "Cannot load model";
+    createCollectors();
+}
 
-    std::vector<Tri> triangles = manager.getModelTriangles();
-    generators::PointSource source;
+std::vector<objects::EnergyCollector *> SceneManager::getEnergyCollectors()
+{
+    std::vector<objects::EnergyCollector *> temp;
+    temp.reserve(_energyCollectors.size());
+    for (const auto &t : _energyCollectors)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-    for (int i = 0; i < 1000000; i++)
+std::vector<objects::TriangleObj *> SceneManager::getModelTriangles()
+{
+    std::vector<objects::TriangleObj *> temp;
+    temp.reserve(_modelTriangles.size());
+    for (const auto &t : _modelTriangles)
     {
-        auto ray = source.generateRay(1, 2);
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-        for (const Tri &triangle : triangles)
+// PRIVATE METHODS
```

Comment:
Remove. The fact that this is private is only relevant in .h file, and here does not give you any additional information. 
There is nothing in the impl. that you would change depending on if this is public or private. What if you decide to make it public? Then you need to change this place as well.

Reply:
done